optaplanner-examples/src/main/java/org/optaplanner/examples/curriculumcourse/optional/score/CurriculumCourseConstraintProvider.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.curriculumcourse.optional.score;
+
+import static org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore.ONE_HARD;
+import static org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore.ofHard;
+import static org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore.ofSoft;
+import static org.optaplanner.core.api.score.stream.ConstraintCollectors.countDistinct;
+import static org.optaplanner.core.api.score.stream.Joiners.equal;
+import static org.optaplanner.core.api.score.stream.Joiners.filtering;
+
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.examples.curriculumcourse.domain.Curriculum;
+import org.optaplanner.examples.curriculumcourse.domain.Lecture;
+import org.optaplanner.examples.curriculumcourse.domain.UnavailablePeriodPenalty;
+import org.optaplanner.examples.curriculumcourse.domain.solver.CourseConflict;
+
+public class CurriculumCourseConstraintProvider implements ConstraintProvider {
```

Comment:
Good rename. The CourseScheduleXmlSolutionFileIO should also be renamed to CurriculumCourseXmlSolutionFileIO